Lib/typing.py

```diff
@@ -801,14 +730,70 @@ def __reduce__(self):
                 args, = args
         return operator.getitem, (origin, args)
 
+    def __mro_entries__(self, bases):
+        if self._name:  # generic version of an ABC or built-in class
+            return super().__mro_entries__(bases)
+        if self.__origin__ is Generic:
+            if Protocol in bases:
+                return ()
+            i = bases.index(self)
+            for b in bases[i+1:]:
+                if isinstance(b, _BaseGenericAlias) and b is not self:
+                    return ()
+        return (self.__origin__,)
+
+
+class _SpecialGenericAlias(_BaseGenericAlias, _root=True):
+    def __init__(self, origin, params, *, inst=True, name=None):
+        if name is None:
+            name = origin.__name__
+        super().__init__(origin, params, inst=inst, name=name)
+        self.__doc__ = f'A generic version of {origin.__module__}.{origin.__qualname__}'
+
+    @_tp_cache
+    def __getitem__(self, params):
+        if not isinstance(params, tuple):
+            params = (params,)
+        msg = "Parameters to generic types must be types."
+        params = tuple(_type_check(p, msg) for p in params)
+        _check_generic(self, params)
+        assert self.__args__ == self.__parameters__
+        return self.copy_with(params)
+
+    def copy_with(self, params):
+        return _GenericAlias(self.__origin__, params,
```

Comment:
Why not `_SpecialGenericAlias(...)`? (If there's a reason, add a comment.)

Reply:
`_SpecialGenericAlias` is for non-parametrized aliases, like `List`. `_GenericAlias` is for parametrized aliases, like `List[int]`. By design, when you you subscribe `_SpecialGenericAlias` you get `_GenericAlias`. `_GenericAlias` is like `GenericAlias` with some additions (for example it checks and converts arguments). `_SpecialGenericAlias` is a producer of `_GenericAlias` (other producer is `_SpecialForm` -- class for `Union`, `ClassVar`, etc).

Reply:
Ah, got it. `List.copy_with(X)` returns `List[X]`.